pkg/server/commands/check_command.go

```diff
@@ -122,9 +150,18 @@ func (c *CheckQuery) Execute(ctx context.Context, params *CheckCommandParams) (*
 
 		return nil, nil, err
 	}
+
+	var datastoreQueryCount uint32
 	if resp != nil {
-		resp.ResolutionMetadata.DatastoreQueryCount = c.datastore.GetMetrics().DatastoreQueryCount
+		datastoreQueryCount = c.datastore.GetMetrics().DatastoreQueryCount
+		resp.ResolutionMetadata.DatastoreQueryCount = datastoreQueryCount
 	}
```

Comment:
I think the time has come to _always_ track `DatastoreQueryCount`, regardless of whether the request errored or not.